File: app/src/main/java/net/veldor/flibustaloader/utils/ZipManager.java
package net.veldor.flibustaloader.utils;

import android.util.Log;

import java.io.BufferedInputStream;
import java.io.BufferedOutputStream;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.util.zip.ZipEntry;
import java.util.zip.ZipOutputStream;

public class ZipManager {
    private static final int BUFFER = 1024;

    public void zip(File[] existentFiles, File outputFile) {
        try {
            BufferedInputStream origin;
            FileOutputStream dest = new FileOutputStream(outputFile);
            ZipOutputStream out = new ZipOutputStream(new BufferedOutputStream(
                    dest));
            byte[] data = new byte[BUFFER];

            for (File file : existentFiles) {
                Log.d("surprise", "Adding: " + file);
                FileInputStream fi = new FileInputStream(file);
                origin = new BufferedInputStream(fi, BUFFER);

                ZipEntry entry = new ZipEntry(file.getName());
                out.putNextEntry(entry);
                int count;

                while ((count = origin.read(data, 0, BUFFER)) != -1) {
                    out.write(data, 0, count);
                }
                origin.close();
            }

            out.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/net/veldor/flibustaloader/utils/ZipManager.kt
package net.veldor.flibustaloader.utils

import android.util.Log
import java.io.*
import java.lang.Exception
import java.util.zip.ZipEntry
import java.util.zip.ZipOutputStream

class ZipManager {
    fun zip(existentFiles: Array<File>, outputFile: File?) {
        try {
            var origin: BufferedInputStream
            val dest = FileOutputStream(outputFile)
            val out = ZipOutputStream(
                BufferedOutputStream(
                    dest
                )
            )
            val data = ByteArray(BUFFER)
            for (file in existentFiles) {
                Log.d("surprise", "Adding: $file")
                val fi = FileInputStream(file)
                origin = BufferedInputStream(fi, BUFFER)
                val entry = ZipEntry(file.name)
                out.putNextEntry(entry)
                var count: Int
                while (origin.read(data, 0, BUFFER).also { count = it } != -1) {
                    out.write(data, 0, count)
                }
                origin.close()
            }
            out.close()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    companion object {
        private const val BUFFER = 1024
    }
}